### PR TITLE
OS-8593 ipmitool update for gcc 14

### PR DESCRIPTION
**Pull request by tsoome**
ipmitool has multiple problems:

- missing prototypes for ipmi_intf_get_max_response_data_size() and ipmi_intf_get_max_request_data_size().
- unused static function fru_area_print_multirec_bloc()
- signed versus unsigned compare
- The if statements which are never true -Wtautological-compare
- string functions are expecting const char * pointer, not unsigned char *

There is also problem about structure with flexible array member used in another structure, but that does emit the warning now, so I did left that one as is.

### CODE REVIEW

File: ipmitool/Patches/gcc-14.patch [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
++					printf("            FRQ: %-9u - min: %-9u - max: %-9u\n",
+ 							freq, min_freq, max_freq);
+ 				}
+ 				printf("\n");
+@@ -3835,6 +3774,8 @@
+ 	}
+ 
+ 	/* Retreive length */
++#if 0
++	/* The if statements here are never true -Wtautological-compare */
```
Comment (danmcd) [inline ipmitool/Patches/gcc-14.patch:218]: If room? Have `XXX SmartOS` there?

File: ipmitool/Patches/gcc-14.patch [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
++#endif
+ 
+ 	*pSize = end - (header.offset.multi * 8);
+ 	*pOffset = (header.offset.multi * 8);
+@@ -4022,6 +3964,8 @@
+ 	end = pFruInfo->size;
+ 
+ 	/* Retreive length */
++#if 0
++	/* The if statements here are never true -Wtautological-compare */
```
Comment (danmcd) [inline ipmitool/Patches/gcc-14.patch:235]: Same here...

**Review by danmcd — COMMENTED** [review]
We *really* need this one to be tested in both gcc10 and gcc14 builds.

**Review by danmcd — APPROVED** [review]
Thanks.  Will need some testing before IA.

**Comment by tsoome** [comment]
> We _really_ need this one to be tested in both gcc10 and gcc14 builds.

I did add sample outputs to the issue.

**Pull request merged**